components/ledger/libs/licence/licence.go

```diff
@@ -0,0 +1,83 @@
+package licence
+
+import (
+	"time"
+
+	"github.com/formancehq/stack/libs/go-libs/logging"
+)
+
+type Licence struct {
+	logger logging.Logger
+
+	jwtToken            string
+	licenceValidateTick time.Duration
+	appStoped           chan struct{}
+
+	// Will be checked in claims (aud claim)
+	serviceName string
+	// kube-system namespace UID, should be in the sub claim
+	clusterID string
+	// Expected issuer, should be in the iss claim
+	expectedIssuer string
+}
+
+func NewLicence(
+	logger logging.Logger,
+	jwtToken string,
+	licenceValidateTick time.Duration,
+	serviceName string,
+	clusterID string,
+	expectedIssuer string,
+) *Licence {
+	return &Licence{
+		logger:              logger,
+		jwtToken:            jwtToken,
+		licenceValidateTick: licenceValidateTick,
+		serviceName:         serviceName,
+		clusterID:           clusterID,
+		expectedIssuer:      expectedIssuer,
+		appStoped:           make(chan struct{}),
+	}
+}
+
+func (l *Licence) run(licenceError chan error) {
+	l.licenceValidateTick = 2 * time.Minute
+	ticker := time.NewTicker(l.licenceValidateTick)
+
+	for {
+		select {
+		case <-l.appStoped:
+			l.logger.Info("Licence check stopped, app stopped")
+			// App is stopped, return
+			return
+
+		case <-ticker.C:
+			l.logger.Info("Licence check started")
+			if err := l.validate(); err != nil {
+				l.logger.Error("Licence check failed", err)
+				licenceError <- err
+				return
+			}
+			l.logger.Info("Licence check passed")
+		}
+	}
+}
+
+func (l *Licence) Start(licenceError chan error) error {
```

Comment:
returning an error here is useless?

Reply:
I need to send an error also for the first time calling it before launching the go routines. I don't want the starts method finishing